code/SecureFileController.php

```diff
@@ -60,18 +61,22 @@ public function handleRequest(SS_HTTPRequest $request, DataModel $model) {
 
                 return $self->sendFile($file);
             } else {
-                if($file instanceof File) {
+                if ($file instanceof File) {
                     // Permission failure
                     Security::permissionFailure($self, 'You are not authorised to access this resource. Please log in.');
                 } else {
-                    // File doesn't exist
-                    $self->setResponse(new SS_HTTPResponse('File Not Found', 404));
+                    if (ClassInfo::class_exists('ErrorPage')) {
+                        $self->setResponse(ErrorPage::response_for(404));
+                    } else {
+                        // File doesn't exist
+                        $this->httpError(404);
```

Comment:
We should `return false;` or something similar here - it's not immediately clear that `httpError` throws an exception to break out of the current execution - and if that method was ever changed to *not* throw an exception, we'd probably want this to still exit instead of falling back to calling `getResponse()`.